<a href="https://colab.research.google.com/github/Arachne01/data-science-2026/blob/main/Pertemuan2_StephenWijaya_250401020158.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [6]:
import pandas as pd
import numpy as np

In [11]:
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

In [14]:
print(df.shape, df.columns.tolist())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Pertanyaan 1, terdapat 12 kolom dan juga 891 bars

In [26]:
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Pertanyaan 2, terdapat 177 values age, 687 values Cabin, dan 2 values Embarked yang masih kosong

In [29]:
wk1 = df[(df['Sex']=="female") & (df['Pclass']==1)]

print(df['Survived'].mean(), wk1['Age'].mean())



0.3838383838383838 34.61176470588235


Pertanyaan 3, ada 38,38% penumpang yang selamat
pertanyaan 4, Rata-rata usia penumpang wanita kelas 1 yang selamat adalah 34

In [24]:
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean'),).round(3)

print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


Pertanyaan 5, kelas 1 memiliki tingkat keselamatan yang paling tinggi

In [32]:
hasil1= df.groupby('Pclass')['Survived'].mean()
print(hasil1)

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64
